presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/CanonicalizeExpressionRewriter.java

```diff
@@ -107,5 +111,29 @@ public Expression rewriteExtract(Extract node, Void context, ExpressionTreeRewri
 
             throw new UnsupportedOperationException("not yet implemented: " + node.getField());
         }
+
+        @Override
+        public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
```

Comment:
This is an optimization. I don't think it belongs to "canonicalize expression". Please implement this as a separate expression rewriting rule.

Reply:
Hmm yeah 

> This is an optimization. I don't think it belongs to "canonicalize expression". Please implement this as a separate expression rewriting rule.

Yeah - I originally suggested this in the task. But sure moving it to expression optimization is ok too.

Reply:
Implemented as separate "FlattenNestedConcat" expression rewrite rule